### PR TITLE
Add ability to upload avatars

### PR DESCRIPTION
**Pull request by reedzyy**

### CODE REVIEW

**Review by SashaShostyr — APPROVED** [review]

**Review by idealasgas — APPROVED** [review]

File: app/views/shared/_user_info.html.slim [hunk 1]
```diff
@@ -1,7 +1,7 @@
 .user-info
   .image
     /! <img src="https://pp.userapi.com/c834401/v834401188/18f70f/x9dYrYMoCzQ.jpg" width="48" height="48" alt="User" /
-    = image_tag("user.png")
+    = image_tag(current_user.user_profile.avatar.icon.url, alt: 'Image') if current_user.user_profile.avatar?
```
Comment (anatoliliotych) [inline app/views/shared/_user_info.html.slim]: where is default avatar? can you move all this magic to model?

File: spec/uploaders/avatar_uploader_spec.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      expect(uploader.icon).to have_dimensions(50, 50)
+    end
+  end
+
+  it 'has the correct format' do
+    expect(uploader).to be_format('JPEG')
+  end
+
+  it 'has the correct size' do
+    expect(uploader.size).to be <=1000000 # bytes
```
Comment (anatoliliotych) [inline spec/uploaders/avatar_uploader_spec.rb]: what about style?

**Review by anatoliliotych — CHANGES_REQUESTED** [review]

File: app/uploaders/avatar_uploader.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+  def store_dir
+    "users/avatars/#{model.id}"
+  end
+
+  def default_url
+    ActionController::Base.helpers.asset_path([version_name, 'avatar.png'].compact.join('_'))
+  end
+
+  # Provide a default URL as a default if there hasn't been a file uploaded:
```
Comment (anatoliliotych) [inline app/uploaders/avatar_uploader.rb]: remove all unneeded comments

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -22,6 +22,6 @@ def user_profile
   end
 
   def user_profile_params
-    params.require(:user).permit(user_profile_attributes: %i(full_name city description))
+    params.require(:user).permit(user_profile_attributes: %i(full_name city description avatar avatar_cache))
```
Comment (anatoliliotych) [inline app/controllers/users_controller.rb]: I don't see where we ignore avatar setting to nil on update

**Review by anatoliliotych — CHANGES_REQUESTED** [review]

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -18,10 +20,10 @@ def update
   private
 
   def user_profile
```
Comment (anatoliliotych) [inline app/controllers/users_controller.rb]: make a before action and set instance variable

File: app/views/shared/_user_info.html.slim [hunk 6]
```diff
@@ -1,7 +1,7 @@
 .user-info
-  .image
-    /! <img src="https://pp.userapi.com/c834401/v834401188/18f70f/x9dYrYMoCzQ.jpg" width="48" height="48" alt="User" /
-    = image_tag("user.png")
+  figure
+    - unless current_user.user_profile.avatar.url.nil?
```
Comment (anatoliliotych) [inline app/views/shared/_user_info.html.slim]: try to not repeat the same, move it to model level

File: app/views/users/show.html.slim [hunk 7]
```diff
@@ -1,17 +1,28 @@
...
+              figure
+                img.img-circle.img-responsive[src=@user_profile.avatar.thumb.url]
+            .col-xs-12.col-sm-8
+              ul.list-group
+                li.list-group-item
+                  | #{@user_profile.full_name}
+                li.list-group-item
+                  | #{@user_profile.city}
+                li.list-group-item
+                  | Some info
```
Comment (anatoliliotych) [inline app/views/users/show.html.slim]: what is some info?

File: app/views/shared/_user_info.html.slim (hunk 6 above)
Comment (anatoliliotych) [reply app/views/shared/_user_info.html.slim]: where default image is shown?

**Review by anatoliliotych — CHANGES_REQUESTED** [review]

File: app/assets/javascripts/user_profile.js [hunk 8]
```diff
@@ -0,0 +1,18 @@
+$(document).ready(function() {
+  $('.birthday_picker').datepicker({
+    dateFormat: 'yy-mm-dd'
+  });
+  function readURL(input) {
```
Comment (anatoliliotych) [inline app/assets/javascripts/user_profile.js]: add empty line

File: app/controllers/users/registrations_controller.rb [hunk 9]
```diff
@@ -1,12 +1,20 @@
 module Users
   class RegistrationsController < Devise::RegistrationsController
-    before_action :configure_sign_up_params, only: %i[create]
-    before_action :configure_account_update_params, only: %i[update]
+    before_action :configure_permitted_parameters, only: %i[create]
+
+    def new
+      build_resource({})
```
Comment (anatoliliotych) [inline app/controllers/users/registrations_controller.rb]: check this

File: app/views/users/edit.html.slim [hunk 10]
```diff
@@ -1,16 +1,73 @@
...
+                = ff.text_field :phone, class: 'form-control'
+        .col-xs-14.col-sm-12
+          = ff.label :description
+          .input-group
+            span.input-group-addon
+              i.material-icons info_outline
+            .form-line
+              = ff.text_field :description, class: 'form-control'
+        .actions
+          = ff.button type: :submit, class: 'btn btn-primary'
```
Comment (anatoliliotych) [inline app/views/users/edit.html.slim]: waves effect

File: app/views/users/show.html.slim [hunk 11]
```diff
@@ -1,17 +1,39 @@
...
+              | Phone: #{@user_profile.phone}
+    .row
+    - if @user_profile.description?
+      h4
+        | Description:
+      p
+        | #{@user_profile.description}
+
+
+
```
Comment (anatoliliotych) [inline app/views/users/show.html.slim]: remove empty lines

**Review by anatoliliotych — CHANGES_REQUESTED** [review]

**Review by anatoliliotych — APPROVED** [review]

**Pull request merged**